src/services/item/schemas.create.ts

```diff
@@ -62,87 +54,94 @@ function itemCreateSchemaFactoryWithSettings<
-  Type.Object(
-    {
-      content: Type.String(),
-      flavor: Type.Optional(Type.Enum(DocumentItemExtraFlavor)),
-      isRaw: Type.Optional(Type.Boolean()),
-    },
-    { additionalProperties: false },
-  ),
+  customType.StrictObject({
+    content: Type.String(),
+    flavor: Type.Optional(Type.Enum(DocumentItemExtraFlavor)),
+    isRaw: Type.Optional(Type.Boolean()),
+  }),
 );
+
+// LINK
 const linkItemCreateSchema = itemCreateSchemaFactoryWithSettings(
   ItemType.LINK,
-  Type.Object(
-    {
-      url: Type.String({ format: 'uri' }),
-      thumbnails: Type.Optional(Type.Array(Type.String())),
-      icons: Type.Optional(Type.Array(Type.String())),
-      html: Type.Optional(Type.String()),
-      description: Type.Optional(Type.String()),
-    },
-    { additionalProperties: false },
-  ),
+  customType.StrictObject({
+    url: Type.String({ format: 'uri' }),
+    thumbnails: Type.Optional(Type.Array(Type.String())),
+    icons: Type.Optional(Type.Array(Type.String())),
+    html: Type.Optional(Type.String()),
+    description: Type.Optional(Type.String()),
+  }),
   Type.Composite(
     [
       settingsSchema,
-      Type.Object(
-        {
-          showLinkIframe: Type.Optional(Type.Boolean()),
-          showLinkButton: Type.Optional(Type.Boolean()),
-        },
-        { additionalProperties: false },
-      ),
+      customType.StrictObject({
+        showLinkIframe: Type.Optional(Type.Boolean()),
+        showLinkButton: Type.Optional(Type.Boolean()),
+      }),
     ],
     {
       additionalProperties: false,
     },
   ),
 );
+
+// FILE
+const fileItemExtra = customType.StrictObject({
+  name: Type.String(),
+});
 const localFileItemCreateSchema = itemCreateSchemaFactory(ItemType.LOCAL_FILE, fileItemExtra);
 const s3FileItemCreateSchema = itemCreateSchemaFactory(ItemType.S3_FILE, fileItemExtra);
 const etherpadItemCreateSchema = itemCreateSchemaFactory(
   ItemType.ETHERPAD,
-  Type.Object({ groupID: Type.String(), padID: Type.String() }, { additionalProperties: false }),
+  customType.StrictObject({ groupID: Type.String(), padID: Type.String() }),
 );
+
+// H5P
 const h5pItemCreateSchema = itemCreateSchemaFactory(
   ItemType.H5P,
-  Type.Object(
-    { contentId: Type.String(), h5pFilePath: Type.String(), contentFilePath: Type.String() },
-    { additionalProperties: false },
-  ),
+  customType.StrictObject({
+    contentId: Type.String(),
+    h5pFilePath: Type.String(),
+    contentFilePath: Type.String(),
+  }),
 );
+
+// SHORTCUT
 const shortcutItemCreateSchema = itemCreateSchemaFactory(
   ItemType.SHORTCUT,
-  Type.Object({ target: customType.UUID() }, { additionalProperties: false }),
+  customType.StrictObject({ target: customType.UUID() }),
 );
 
 export const create = {
+  operationId: 'createItem',
+  tags: ['item'],
+  summary: 'Create item',
+  description:
+    'Create item, whose possible types are folder, app, document, embeddedLink, localFile, s3File, etherpad, h5p and shortcut.',
```

Comment:
This could be generated from the ItemType enum ?

Reply:
Potentially we will remove it, so I let it be.